StRoot/StETofCalibMaker/StETofCalibMaker.cxx

```diff
@@ -2609,3 +2667,275 @@ StETofCalibMaker::writeHistograms()
         LOG_INFO << "histogram file name is empty string --> cannot write histograms" << endm;
     }
 }
+
+//--------------------------------------------------------------------------------------------------------------------
+
+void StETofCalibMaker::readGet4State(int fileNr, short forward){
+
+   bool fileZero = false;
+
+   //Clean up last entry first
+   for(int i =0; i< eTofConst::nGet4sInSystem; i++){
+     mStateVec[i].clear();
+     mStartVec[i].clear();
+     mGet4StateMap[i] = 0;
+   }
+   mStateMapStart=0;
+   mStateMapStop=0;
+   mDbEntryStop=0;
+   mMasterStartVec.clear();
+   mMasterStartVec.resize(0);
+
+   std::vector< unsigned long int > intVec;
+   
+   //first read
+    if(forward == 0) mGlobalCounter = 1;
+    //jump forward
+    else if(forward > 0) mGlobalCounter++;
+    //jump backward
+    else mGlobalCounter--; // forward < 0
+    
+    if(mGlobalCounter == 0){
+      mGlobalCounter++;
+      fileZero = true;
+    }
+    
+    if(mFileNameGet4State.empty()){
+         
+      TDataSet* dbDataSet = GetDataBase( "Calibrations/etof/etofGet4State" );
+      const int intsPerEntry = 1000000;
+      
+      St_etofGet4State* etofStateMap = static_cast< St_etofGet4State* > ( dbDataSet->Find( "etofGet4State" ) );
+      if( !etofStateMap ) {
+	LOG_ERROR << "unable to get the get4 state map from the database" << endm;	  
+	return;
+      }
```

Comment:
My guess is that some of the test jobs are crashing because the `GetDataBase()` call returned 0.
```suggestion
      TDataSet* dbDataSet = GetDataBase( "Calibrations/etof/etofGet4State" );
      if( ! dbDataSet ) {
	LOG_ERROR << "unable to get the get4 state map database" << endm;	  
	return;
      }
      const int intsPerEntry = 1000000;
      
      St_etofGet4State* etofStateMap = static_cast< St_etofGet4State* > ( dbDataSet->Find( "etofGet4State" ) );
      if( !etofStateMap ) {
	LOG_ERROR << "unable to get the get4 state map from the database" << endm;	  
	return;
      }
```